src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2153,32 +2153,48 @@ LogicalResult ONNXSplitOp::inferShapes() {
 //===----------------------------------------------------------------------===//
 
 LogicalResult ONNXFlattenOp::inferShapes() {
-  assert(axis() == 1 && "ONNXFlattenOp can only handle axis=1 for now");
   auto inTy = input().getType().dyn_cast<ShapedType>();
   if (!inTy) {
     return emitOpError("Input is a non-shaped type");
   }
-  auto outTy = output().getType().dyn_cast<ShapedType>();
-  if (!outTy) {
-    return emitOpError("Output is a non-shaped type");
+
+  auto axisValue = axis();
+  auto inputShape = inTy.getShape();
+  auto inputRank = inputShape.size();
+  if (axisValue < -1 * (int64_t)inputRank || axisValue > (int64_t)inputRank) {
+    return emitOpError("ONNXFlattenOP: axis() value is out of range");
   }
 
-  // TODO(tjingrant): Seems like we can also fairly easily support the case
-  //                  where the batch dimension is dynamic
-  if (!outTy.hasStaticShape()) {
-    auto inShape = inTy.getShape();
-    assert(inShape.size() >= 1 && "ONNXFlattenOp inShape.size() should be > 0");
-    uint64_t outDim = 1;
-    for (auto it = inShape.begin() + 1; it < inShape.end(); it++) {
-      outDim *= *it;
+  SmallVector<int64_t, 2> dims;
+
+  // Negative axis is counting dimension from back
+  if (axisValue < 0)
+    axisValue = inputRank + axisValue + 1;
+
+  // Determine the size of the first dimension of output
+  uint64_t firstDim = 1;
+  for (auto i = 0; i < axisValue; i++) {
+    if (inputShape[i] == -1) {
+      firstDim = -1;
```

Comment:
Seems like you are assigning a negative value to an unsigned variable (firstDim is declared as uint64_t).

Reply:
fixed